// @caido/tailwindcss@0.0.1/dist/tailwindcss.js
function v(r) {
  return r && r.__esModule && Object.prototype.hasOwnProperty.call(r, "default") ? r.default : r;
}
var t = {}, h = {};
(function(r) {
  Object.defineProperty(r, "__esModule", {
    value: !0
  }), Object.defineProperty(r, "default", {
    enumerable: !0,
    get: function() {
      return l;
    }
  });
  function c(a, e) {
    return {
      handler: a,
      config: e
    };
  }
  c.withOptions = function(a, e = () => ({})) {
    const s = function(n) {
      return {
        __options: n,
        handler: a(n),
        config: e(n)
      };
    };
    return s.__isOptionsFunction = !0, s.__pluginFunction = a, s.__configFunction = e, s;
  };
  const l = c;
})(h);
(function(r) {
  Object.defineProperty(r, "__esModule", {
    value: !0
  }), Object.defineProperty(r, "default", {
    enumerable: !0,
    get: function() {
      return a;
    }
  });
  const c = /* @__PURE__ */ l(h);
  function l(e) {
    return e && e.__esModule ? e : {
      default: e
    };
  }
  const a = c.default;
})(t);
let u = t;
var o = (u.__esModule ? u : { default: u }).default;
const i = /* @__PURE__ */ v(o), f = i(() => {
}, {
  theme: {
    extend: {
      colors: {
        primary: {
          100: "hsl(var(--c-primary-100))",
          200: "hsl(var(--c-primary-200))",
          300: "hsl(var(--c-primary-300))",
          400: "hsl(var(--c-primary-400))",
          500: "hsl(var(--c-primary-500))",
          600: "hsl(var(--c-primary-600))",
          700: "hsl(var(--c-primary-700))",
          800: "hsl(var(--c-primary-800))",
          900: "hsl(var(--c-primary-900))"
        },
        secondary: {
          100: "hsl(var(--c-secondary-100))",
          200: "hsl(var(--c-secondary-200))",
          300: "hsl(var(--c-secondary-300))",
          400: "hsl(var(--c-secondary-400))",
          500: "hsl(var(--c-secondary-500))",
          600: "hsl(var(--c-secondary-600))",
          700: "hsl(var(--c-secondary-700))",
          800: "hsl(var(--c-secondary-800))",
          900: "hsl(var(--c-secondary-900))"
        },
        danger: {
          100: "hsl(var(--c-danger-100))",
          200: "hsl(var(--c-danger-200))",
          300: "hsl(var(--c-danger-300))",
          400: "hsl(var(--c-danger-400))",
          500: "hsl(var(--c-danger-500))",
          600: "hsl(var(--c-danger-600))",
          700: "hsl(var(--c-danger-700))",
          800: "hsl(var(--c-danger-800))",
          900: "hsl(var(--c-danger-900))"
        },
        info: {
          100: "hsl(var(--c-info-100))",
          200: "hsl(var(--c-info-200))",
          300: "hsl(var(--c-info-300))",
          400: "hsl(var(--c-info-400))",
          500: "hsl(var(--c-info-500))",
          600: "hsl(var(--c-info-600))",
          700: "hsl(var(--c-info-700))",
          800: "hsl(var(--c-info-800))",
          900: "hsl(var(--c-info-900))"
        },
        success: {
          100: "hsl(var(--c-success-100))",
          200: "hsl(var(--c-success-200))",
          300: "hsl(var(--c-success-300))",
          400: "hsl(var(--c-success-400))",
          500: "hsl(var(--c-success-500))",
          600: "hsl(var(--c-success-600))",
          700: "hsl(var(--c-success-700))",
          800: "hsl(var(--c-success-800))",
          900: "hsl(var(--c-success-900))"
        },
        surface: {
          0: "hsl(var(--c-surface-0))",
          200: "hsl(var(--c-surface-200))",
          300: "hsl(var(--c-surface-300))",
          400: "hsl(var(--c-surface-400))",
          500: "hsl(var(--c-surface-500))",
          600: "hsl(var(--c-surface-600))",
          700: "hsl(var(--c-surface-700))",
          800: "hsl(var(--c-surface-800))",
          900: "hsl(var(--c-surface-900))"
        }
      }
    }
  }
});
export {
  f as default
};
